app/src/main/java/org/wikipedia/talk/template/TalkTemplatesTextInputDialog.kt

```diff
@@ -122,6 +129,17 @@ class TalkTemplatesTextInputDialog constructor(context: Context,
         }
     }
 
+    private fun sendPatrollerExperienceEvent(action: String) {
+        var activity: Context? = context
+        while (activity !is Activity && activity is ContextWrapper) {
+            activity = activity.baseContext
+        }
+        activity?.let {
+            PatrollerExperienceEvent.logAction(action, if (it is TalkReplyActivity) "pt_warning_messages"
+            else "pt_templates")
+        }
+    }
```

Comment:
Would you be able to try the following one?
```
class TalkTemplatesTextInputDialog constructor(private val activity: Activity,
                                               positiveButtonText: Int = R.string.text_input_dialog_ok_button_text,
                                               negativeButtonText: Int = R.string.text_input_dialog_cancel_button_text)
```
Change the constructor from `context: Context` to `private val activity: Activity` so you can skip additional logic to check the activity and conversion. 

And this can be simplified to:
```
    private fun sendPatrollerExperienceEvent(action: String) {
            PatrollerExperienceEvent.logAction(action, if (activity is TalkReplyActivity) "pt_warning_messages"
            else "pt_templates")
    }
```